### ISSUE TITLE
dotnet test does no longer accept a project path after updating to NET 7.0.101

### ISSUE THREAD

**Issue by campersau** [dotnet/sdk#29543]
### Describe the bug
After updating NET 7.0.100 to NET 7.0.101 `dotnet test` does no longer accept a project path as an argument.

### To Reproduce

Sample test project:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.0" />
    <PackageReference Include="xunit" Version="2.4.2" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.5">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>

</Project>
```

```c#
using Xunit;

namespace ClassLibrary1
{
    public class Class1
    {
        [Fact]
        public void Test()
        {

        }
    }
}
```

With NET 7.0.101 this does not work
```
C:\>dotnet --version
7.0.101
C:\>dotnet test C:\Users\Administrator\Desktop\ClassLibrary1.csproj
Befehl oder Argument "C:\Users\Administrator\Desktop\ClassLibrary1.csproj" nicht erkannt
```

With NET 7.0.100 this worked

```
C:\>dotnet --version
7.0.100
C:\>dotnet test C:\Users\Administrator\Desktop\ClassLibrary1.csproj
  Determining projects to restore...
  All projects are up-to-date for restore.
  ClassLibrary1 -> C:\Users\Administrator\Desktop\bin\Debug\net7.0\ClassLibrary1.dll
Test run for C:\Users\Administrator\Desktop\bin\Debug\net7.0\ClassLibrary1.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.4.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration: < 1 ms - ClassLibrary1.dll (net7.0)
```

This also affects the built in dotnet command in Azure Dev Ops.
![image](https://user-images.githubusercontent.com/4009570/207575091-949476f2-a553-4cf9-8f0e-0981b5ea8ea4.png)

### Workarounds

- Set the environment variable `DOTNET_CLI_UI_LANGUAGE=en-US`.

- Running `donet test` directly in the project directory does work:

```
C:\Users\Administrator\Desktop>dotnet test
  Wiederherzustellende Projekte werden ermittelt...
  Alle Projekte sind für die Wiederherstellung auf dem neuesten Stand.
  ClassLibrary1 -> C:\Users\Administrator\Desktop\bin\Debug\net7.0\ClassLibrary1.dll
Testlauf für "C:\Users\Administrator\Desktop\bin\Debug\net7.0\ClassLibrary1.dll" (.NETCoreApp,Version=v7.0)
Microsoft (R) Testausführungs-Befehlszeilentool Version 17.4.0 (x64)
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Die Testausführung wird gestartet, bitte warten...
Insgesamt 1 Testdateien stimmten mit dem angegebenen Muster überein.

Bestanden!   : Fehler:     0, erfolgreich:     1, übersprungen:     0, gesamt:     1, Dauer: < 1 ms - ClassLibrary1.dll (net7.0)
```

- Running `donet test` referencing a DLL or exe does work:

```
C:\Users\Administrator\Desktop>dotnet test C:\Users\Administrator\Desktop\bin\Debug\net7.0\ClassLibrary1.dll
  Wiederherzustellende Projekte werden ermittelt...
  Alle Projekte sind für die Wiederherstellung auf dem neuesten Stand.
  ClassLibrary1 -> C:\Users\Administrator\Desktop\bin\Debug\net7.0\ClassLibrary1.dll
Testlauf für "C:\Users\Administrator\Desktop\bin\Debug\net7.0\ClassLibrary1.dll" (.NETCoreApp,Version=v7.0)
Microsoft (R) Testausführungs-Befehlszeilentool Version 17.4.0 (x64)
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Die Testausführung wird gestartet, bitte warten...
Insgesamt 1 Testdateien stimmten mit dem angegebenen Muster überein.

Bestanden!   : Fehler:     0, erfolgreich:     1, übersprungen:     0, gesamt:     1, Dauer: < 1 ms - ClassLibrary1.dll (net7.0)
```

### Further technical details
```
C:\>dotnet --info
.NET SDK:
 Version:   7.0.101
 Commit:    bb24aafa11

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.20348
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.101\

Host:
  Version:      7.0.1
  Architecture: x64
  Commit:       97203d38ba

.NET SDKs installed:
  7.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 7.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found
```

**Comment by stefannikolei** [comment]
Probably introduced with this pr: https://github.com/dotnet/sdk/pull/27961

This is blocking all of my CI runs

**Comment by matjankow** [comment]
Seems to be the same moving from .net SDK 6.0.403 to .net 6.0.404

**Comment by eluchsinger** [comment]
Azure DevOps Pipelines stopped working for me aswell.

**Comment by meriturva** [comment]
Same here with teamcity!

**Comment by jansapp** [comment]
Same here

**Comment by MarcoRossignoli** [comment]
Thanks for reporting it, we're investigating.

**Comment by MarcoRossignoli** [comment]
Can you tell me if the same is happening for non localized version(en-us one)?
I see non default language usage.

**Comment by MarcoRossignoli** [comment]
@campersau can you enable verbose logging and retry the command?

```bash
C:\>set DOTNET_CLI_VSTEST_TRACE=1
C:\>dotnet test...
...
[dotnet test - 12/14/2022 4:17:29 PM]Logging to console
[dotnet test - 12/14/2022 4:17:29 PM +00:00]Argument list: 'C:\...'
[dotnet test - 12/14/2022 4:17:29 PM +00:00]MSBuild args from forwarded options: C:\...
```
You should see some verbose output `[dotnet test...]`

**Comment by campersau** [comment]
@MarcoRossignoli here is the german output with `DOTNET_CLI_VSTEST_TRACE=1`

```
C:\>set DOTNET_CLI_VSTEST_TRACE=1
C:\>dotnet test C:\Users\Administrator\Desktop\ClassLibrary1.csproj
[dotnet test - 14.12.2022 16:29:07]Logging to console
[dotnet test - 14.12.2022 16:29:07 +00:00]Argument list: 'C:\Users\Administrator\Desktop\ClassLibrary1.csproj'
Befehl oder Argument "C:\Users\Administrator\Desktop\ClassLibrary1.csproj" nicht erkannt
Description:
  .NET-Testtreiber

Usage:
  dotnet test [options] [[--] <additional arguments>...]]

...
```

---

When changing the Windows display language to `English (United States)` it works:
![image](https://user-images.githubusercontent.com/4009570/207657205-59814750-1a26-4955-8a9c-dbe4c5e79480.png)

```
C:\>set DOTNET_CLI_VSTEST_TRACE=1
C:\>dotnet test C:\Users\Administrator\Desktop\ClassLibrary1.csproj
[dotnet test - 12/14/2022 4:50:51 PM]Logging to console
[dotnet test - 12/14/2022 4:50:51 PM +00:00]Argument list: 'C:\Users\Administrator\Desktop\ClassLibrary1.csproj'
[dotnet test - 12/14/2022 4:50:51 PM +00:00]MSBuild args from forwarded options: C:\Users\Administrator\Desktop\ClassLibrary1.csproj
[dotnet test - 12/14/2022 4:50:51 PM +00:00]Root variable set VSTEST_WINAPPHOST_DOTNET_ROOT:C:\Users\Administrator\Downloads\dotnet-sdk-7.0.101-win-x64
[dotnet test - 12/14/2022 4:50:51 PM +00:00]Starting test using MSBuild with arguments '-maxcpucount -verbosity:m -restore -target:VSTest -nodereuse:false -nologo C:\Users\Administrator\Desktop\ClassLibrary1.csproj -property:VSTestArtifactsProcessingMode=collect -property:VSTestSessionCorrelationId=2416_5b9e1843-62f7-4ff5-91e0-2b5c8446d39b -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Users\Administrator\Downloads\dotnet-sdk-7.0.101-win-x64\sdk\7.0.101\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Users\Administrator\Downloads\dotnet-sdk-7.0.101-win-x64\sdk\7.0.101\dotnet.dll' custom MSBuild path '' norestore 'False'
  Determining projects to restore...
  All projects are up-to-date for restore.
  ClassLibrary1 -> C:\Users\Administrator\Desktop\bin\Debug\net7.0\ClassLibrary1.dll
Test run for C:\Users\Administrator\Desktop\bin\Debug\net7.0\ClassLibrary1.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.4.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration: < 1 ms - ClassLibrary1.dll (net7.0)
[dotnet test - 12/14/2022 4:50:55 PM +00:00]No artifact found, post-processing won't run.
```

**Comment by MarcoRossignoli** [comment]
Thanks I can confirm also on mine, something is wrong with the default parsing of arguments on non eng-us

**Comment by MarcoRossignoli** [comment]
@baronfel what you think?

**Comment by marcpopMSFT** [comment]
If it's failing in parsing, that would likely be a System.CommandLine issue. You could file an issue here: https://github.com/dotnet/command-line-api but @baronfel @adamsitnik may be able to help.

**Comment by baronfel** [comment]
@marcpopMSFT yep! I'll be digging in more shortly to help direct.

**Comment by campersau** [comment]
Setting the environment variable [`DOTNET_CLI_UI_LANGUAGE=en-US`](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables#dotnet_cli_ui_language) does also fix the issue.

**Comment by baronfel** [comment]
Copying the [latest PR update](https://github.com/dotnet/sdk/pull/29559#issuecomment-1359593200) here:

>I think we're in a good space now with the algorithm and the tests for this check.  I'll want to get another check from @dsplaisted to make sure he's comfortable with the implementation, but then we're ready to go.
>
>A note about the timing of this fix: we are already beyond the merge window for the January servicing release of .NET 7 (which would be 7.0.102) due to the holiday season. As a result, this fix (when approved and merged) will only be available in the Februrary servicing release at earliest (so 7.0.103). This feels really bad, and isn't a great response time given the very fast time to reporting that users had, but it's the best I can do.
>
>In the meantime, the best thing I can suggest for users that are hitting this error is to force the CLI to the English local for `dotnet test`. The easiest way to do that is to set the `DOTNET_CLI_UI_LANGUAGE` environment variable in your shell to `en-US` for the duration of the test command. e.g. in Bash you'd run `DOTNET_CLI_UI_LANGUAGE='en-us' dotnet test .....`. Full documentation for this environment variable can be found [here](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables#dotnet_cli_ui_language).

**Comment by stefannikolei** [comment]
So these means we need to tweak our build process to get the security fixes in (which were released with 7.0.101) otherwise our apps are vulnerable till at least February?

**Comment by baronfel** [comment]
@stefannikolei unfortunately yes, that's correct. It's very unfortunate that this problem was surfaced at this specific time - we usually have a much larger change window before lockdown/stabilization/testing phases.

**Comment by Starkie** [comment]
We've encountered this error today after updating our build agents. While waiting for the servicing release, we'll try to make do with the workaround. Thanks for the fast response!

On the other hand, something similar happened in the previous release that also broke our CI setup. (https://github.com/microsoft/vstest/issues/4014) It's making us wary of updating to newer releases of the .NET SDK. 😢 Would it be posible to provide something akin to hotfix releases when these kind of errors occur?

**Comment by meriturva** [comment]
Seems a new version of .net 7 has been released (7.0.2). I will test it on Friday on one or two CI agents on our infrastructure, but I would like to know about other experiences here. Who has the opportunity to test it?

**Comment by MarcoRossignoli** [comment]
@meriturva the fix for this issue will be into `7.0.103` not `7.0.2` as described here https://github.com/dotnet/sdk/issues/29543#issuecomment-1359646795 by @baronfel

**Comment by kenan-kajkus** [comment]
@meriturva just tested it, does not work

**Comment by marcpopMSFT** [comment]
Per @baronfel's comment above, this fix is expected to be released in 7.0.103 in February
